medusa/backup_node.py

```diff
@@ -65,6 +65,15 @@ def __init__(self, *, node_backup, differential_mode, enable_md5_checks,
         self._storage_config = storage_config
         self._enable_md5_checks = enable_md5_checks
 
+    def _retain_2i_subdir(self, path):
+        # Secondary indexes are stored as subdirectories to the base table, starting with a dot.
+        # In order to avoid mixing 2i sstables with the base table sstables, the file name isn't enough
+        # to perform the comparison on differential backups. We need to retain the subdir name for 2i tables.
+        if path.parts[-2].startswith('.'):
+            return os.path.join(path.parts[-2], path.parts[-1])
+        else:
+            return path.name
```

Comment:
Checking, is this for backwards compatibility to return the `path.name`?  If it isn't there for some reason due to failure condition, etc. do we need to log that fact?

Reply:
The presence of the dot at the start of the directory name means it's a secondary index and we need to retain the directory name + the file name (`path.parts[-1]` and `path.name` are the same thing).
If that part doesn't start with a dot, it means we're dealing with the base table sstable files, and not the secondary index. In this case, we just need the file name, not the directory it's in.

Reply:
👍🏻